### PR TITLE
Skip ceph-file-controller-detect-version pod check in test_add_capacity[10]

### LINKED ISSUE #10748: Skip check on temp. pods like ceph-file-controller-detect-version-jdmd4 for test_add_capacity[10]
ODF 4.17 failure- 

E               ocs_ci.ocs.exceptions.CommandFailed: Error during execution of command: oc --kubeconfig /home/jenkins/current-cluster-dir/openshift-cluster-dir/auth/kubeconfig -n openshift-storage get Pod ceph-file-controller-detect-version-jdmd4 -n openshift-storage.
E               Error is Error from server (NotFound): pods "ceph-file-controller-detect-version-jdmd4" not found

In code:
  restart_count_after = pod_helpers.get_pod_restarts_count(
            config.ENV_DATA["cluster_namespace"]
        )

### PR DESCRIPTION
**Pull request by am-agrawa**
fixes #10748

### CODE REVIEW

**Comment by am-agrawa** [comment]
Passed here- https://url.corp.redhat.com/d015866
Conf- deployment/vsphere/upi_1az_rhcos_vsan_3m_3w.yaml
ODF GA'ed (CSV found: odf-operator.v4.17.0-rhodf)

File: ocs_ci/ocs/resources/pod.py [hunk 1]
```diff
@@ -2333,10 +2333,13 @@ def get_pod_restarts_count(
...
     ocp_pod_obj = OCP(kind=constants.POD, namespace=namespace)
     for p in list_of_pods:
-        # we don't want to compare osd-prepare and canary pods as they get created freshly when an osd need to be added.
+        # we don't want to compare osd-prepare and canary pods as they get
+        # created freshly when an osd need to be added. Also skip check on ceph-file-controller-detect-version
+        # as it's a temp. pod.
         if (
             "rook-ceph-osd-prepare" not in p.name
             and "rook-ceph-drain-canary" not in p.name
+            and "ceph-file-controller-detect-version" not in p.name
```
Comment (AviadP) [inline ocs_ci/ocs/resources/pod.py:2342]: imo, this if got too many condition, i would consider something like this:
```
exclude_names = (
    "rook-ceph-osd-prepare",
    "rook-ceph-drain-canary",
    "ceph-file-controller-detect-version"
)

if all(exclude_name not in p.name for exclude_name in exclude_names):
```

Comment (am-agrawa) [reply ocs_ci/ocs/resources/pod.py]: good idea, updated

**Comment by am-agrawa** [comment]
Passed again after the feedback implementation- https://url.corp.redhat.com/f45f2da

**Review by AviadP — APPROVED** [review]

**Review by ebenahar — APPROVED** [review]

**Pull request merged**